src/js/integrations/modulesloader.ts

```diff
@@ -0,0 +1,35 @@
+import { Event,EventProcessor, Integration } from '@sentry/types';
+
+import { NATIVE } from '../wrapper';
+
+/** Loads runtime JS modules from prepared file. */
+export class ModulesLoader implements Integration {
+  /**
+   * @inheritDoc
+   */
+  public static id: string = 'ModulesLoader';
+
+  /**
+   * @inheritDoc
+   */
+  public name: string = ModulesLoader.id;
+
+  /**
+   * @inheritDoc
+   */
+  public setupOnce(addGlobalEventProcessor: (e: EventProcessor) => void): void {
+    let isSetup = false;
+    let modules: Record<string, string> | null;
+
+    addGlobalEventProcessor(async (event: Event) => {
+      if (!isSetup) {
+        modules = await NATIVE.fetchModules();
+        isSetup = true;
+      }
+      if (modules) {
+        event.modules = modules;
```

Comment:
I'd append the items rather than replace it.
If one set their own `modules`, you'd overwrite them.